GitUI/CommandsDialogs/AheadBehindHelper.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using GitCommands;
+using GitUIPluginInterfaces;
+
+namespace GitUI.CommandsDialogs
+{
+    public class AheadBehindHelper
+    {
+        public struct AheadBehindData
+        {
+            public string Branch { get; set; }
+            public string AheadCount { get; set; }
+            public string BehindCount { get; set; }
+
+            public string ToDisplay()
+            {
+                return (!string.IsNullOrEmpty(AheadCount) ? AheadCount + "↑" : string.Empty)
+                       + (!string.IsNullOrEmpty(AheadCount) && !string.IsNullOrEmpty(BehindCount) ? " " : string.Empty)
+                       + (!string.IsNullOrEmpty(BehindCount) ? BehindCount + "↓" : string.Empty);
+            }
+        }
+
+        private readonly IGitModule _gitModule;
+
+        private readonly Regex _aheadBehindRegEx =
+            new Regex(@"^\[(ahead (?<ahead>\d+))?(, )?(behind (?<behind>\d+))?\] (?<branch>.*)$",
+                RegexOptions.Compiled | RegexOptions.Multiline);
+
+        public AheadBehindHelper(IGitModule gitModule)
```

Comment:
I think in this instance we can avoid passing `IGitModule` - it is really used to execute a git command, that in turn is facilitated by an instance of `IExecutable` on an instance of `GitModule`.